pkg/webui/components/rights-group/index.js

```diff
@@ -94,214 +92,155 @@ const computeProps = props => {
-    /**
-     * Whether the entity has a pseudo right that the current use does not
-     * have.
-     */
-    hasOutOfOwnScopePseudoRight: PropTypes.bool.isRequired,
-    /** Whether the entity has a pseudo right granted. */
-    hasPseudoRightGranted: PropTypes.bool.isRequired,
-    /**
-     * The intl object provided by injectIntl of react-intl, used to translate
-     * messages.
-     */
-    intl: PropTypes.shape({
-      formatMessage: PropTypes.func.isRequired,
-    }).isRequired,
-    /** The Blur event hook. */
-    onBlur: PropTypes.func,
-    /** The Change event hook. */
-    onChange: PropTypes.func,
-    /** A list of rights that are outside the scope of the current user. */
-    outOfOwnScopeIndividualRights: PropTypes.rights.isRequired,
-    /** The pseudo right literal comprising all other rights. */
-    pseudoRight: PropTypes.oneOfType([PropTypes.string, PropTypes.arrayOf(PropTypes.string)]),
-    /** The rights value. */
-    value: PropTypes.rights.isRequired,
-  }
-
-  static defaultProps = {
-    className: undefined,
-    disabled: false,
-    onBlur: () => null,
-    onChange: () => null,
-    pseudoRight: undefined,
-    derivedPseudoRight: undefined,
-  }
-
-  state = {
-    individualRightValue: [],
-  }
-
-  static getDerivedStateFromProps(props, state) {
-    const { individualRightValue: oldIndividualRightValue } = state
-    const { value, hasPseudoRightGranted } = props
-
-    // Store the individual right values when the grant type is changed to
-    // pseudo right.
-    const individualRightValue = !hasPseudoRightGranted ? value : oldIndividualRightValue
-
-    return { individualRightValue }
-  }
-
-  @bind
-  handleChangeAll(event) {
-    const { onChange, outOfOwnScopeIndividualRights, derivedRights } = this.props
-    const { checked } = event.target
-
-    let value
-
-    if (checked) {
-      // Fill up with individual rights.
-      value = [...derivedRights]
-    } else {
-      // On uncheck, leave out of scope rights checked, if present.
-      value = [...outOfOwnScopeIndividualRights]
-    }
-
-    onChange(value)
-  }
-
-  @bind
-  handleChange(val) {
-    const { onChange } = this.props
-    const value = Object.keys(val).filter(right => val[right])
-
-    onChange(value)
-  }
-
-  @bind
-  handleGrantTypeChange(val) {
-    const { onChange, pseudoRight } = this.props
-    const { individualRightValue } = this.state
-
-    if (val === 'pseudo') {
-      onChange(Array.isArray(pseudoRight) ? pseudoRight : [pseudoRight])
-    } else {
-      onChange(individualRightValue)
-    }
+const RightsGroup = props => {
+  const {
+    className,
+    derivedPseudoRight,
+    derivedRights,
+    disabled,
+    entityTypeMessage,
+    grantType,
+    hasOutOfOwnScopePseudoRight,
+    hasPseudoRightGranted,
+    onBlur,
+    onChange,
+    outOfOwnScopeIndividualRights,
+    pseudoRight,
+    value,
+  } = useComputedProps(computeProps, props)
```

Comment:
This doesn't really make sense. You might just as well do:
```js
const {…} = computeProps(props)
```

`computeProps` only uses three props: `value`, `pseudoRight`, and `rights`. It would be best to make use of `useMemo` here to make these expensive computations only when these props change.

```js
const {
    outOfOwnScopeIndividualRights,
    hasOutOfOwnScopePseudoRight,
    derivedPseudoRight,
    derivedRights,
    hasPseudoRightGranted,
    grantType,
} = useMemo(computeProps, [value, pseudoRight, rights])
```
This could possibly also be moved into a hook to use it like this:
```js
const {
    outOfOwnScopeIndividualRights,
    hasOutOfOwnScopePseudoRight,
    derivedPseudoRight,
    derivedRights,
    hasPseudoRightGranted,
    grantType,
} = useDerivedRightProps(value, pseudoRight, rights)